lib/src/credentials.dart

```diff
@@ -139,6 +139,8 @@ extension CredentialsInternal on Credentials {
   }
 
   RealmAppCredentialsHandle get handle => _handle;
+
+  AuthProviderType readProviderFromCore() => realmCore.userGetCredentialsProviderType(this);
```

Comment:
```suggestion
  /// This should be only used for testing
  AuthProviderType get credentialsProviderType => realmCore.userGetCredentialsProviderType(this);
```
we have similar comment elsewhere and also suggested a change to a getter and the rename